api/v1alpha1/volumereplicationgroup_types.go

```diff
@@ -18,53 +18,32 @@ package v1alpha1
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	vrv1alpha1 "github.com/kube-storage/volume-replication-operator/api/v1alpha1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
-// VolumeTakeoverControlType -- takeover control when desiredCluster is different from affinedCluster
-type VolumeTakeoverControlType string
-
-// VolumeTakeoverControlType definitions
-const (
-	// Force promote the volume in a WAN DR setting
-	ForcePromote VolumeTakeoverControlType = "ForcePromote"
-)
-
-// VolumeReplicationGroupSpec defines the desired state of VolumeReplicationGroup
+// VolumeReplicationGroup (VRG) spec declares the desired replication class
+// and replication state of all the PVCs owned by a given application, which
+// are identified via the given application label selector.  The VRG operator
+// creates children VolumeReplication (VR) CRs for each PVC of the application,
+// with the desired replication class and replication image state (primary or
+// secondary).
 type VolumeReplicationGroupSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
 	// Name of the application to replicate
-	ApplicationName string `json:"applicationName"`
-
-	// Application label that is used to identify its PVs
-	ApplicationLabels metav1.LabelSelector `json:"applicationLabels"`
-
-	// Cluster ID in ClusterPeers that has best storage performance affinity for the application
-	AffinedCluster string `json:"affinedCluster"`
-
-	// Desired cluster that should takeover the application from current active cluster.
-	// May be set either to:
-	// - secondary cluster ID (to migrate application or takeover application in case of a disaster)
-	// - nil (if application is not active in the affined cluster, takeback to affined cluster)
-	// +optional
-	DesiredCluster string `json:"desiredCluster,omitempty"`
+	ApplicationName string `json:"applicationName,omitempty"`
 
-	// Volume Takeover Control: ForcePromote
-	// +optional
-	VolumeTakeoverControl VolumeTakeoverControlType `json:"volumeTakeoverControl,omitempty"`
+	// Application label selector that is used to identify all its PVCs
+	ApplicationLabelSelector metav1.LabelSelector `json:"applicationLabelSelector"`
```

Comment:
Torn between naming this with an `Application` anything prefix.

The selector applies to PVCs, so naming this as `PVCSelector` may make it more obvious.

Reply:
Or may be ApplicationResourceSelector? In future if we need to list some other resource apart from PVCs for the application using the label selector, this might be a better name?

Reply:
Both good points.  As a compromise, how about ReplicationResourceSelector, which also adds clarity that these are the resources VRG will be replicating?

Reply:
For now, going with PVCSelector that has two votes (Shyam and Benamar).

Reply:
Ack! A more generic selector may need to be combined with the type. For now we can leave it explicit that this selects PVCs.